src/main/java/ru/mingun/kaitai/struct/tree/SimpleNode.java

```diff
@@ -37,14 +37,20 @@ public class SimpleNode extends ChunkNode {
   /** Parsed value of non-constructed type. */
   private final Object value;
 
-  SimpleNode(String name, Object value, ChunkNode parent, long offset, long start, long end) {
+  /** Replace value.getClass() when value==null. */
+  private final Class<?> classOfValue;
```

Comment:
```suggestion
  /** Class of property, represented by the getter return value. */
  private final Class<?> valueClass;
```
`value.getClass()` is slightly incorrect, because it is dynamic type, but in fact the static type is stored here. For example `ArrayList` vs `List`